src/main/java/io/vertx/gsoc2023/qotd/QuoteOfTheDayVerticle.java

```diff
@@ -44,14 +46,43 @@ private PgPool setupPool(JsonObject config) {
 
   private Router setupRouter() {
     var router = Router.router(vertx);
+    setupGet(router);
+    setupPost(router);
+    return router;
+  }
+
+  private void setupPost(Router router) {
+    router.post("/quotes")
+      .consumes("application/json")
+      .handler(BodyHandler.create())
+      .handler(ctx -> {
+        JsonObject body = ctx.body().asJsonObject();
+        if (body != null && body.containsKey("text")) {
```

Comment:
If the `author` contains `text` then the JSON object will be considered valid

Reply:
It checks that the JSON has a key called "text", not that the JSON contains the word "text". The following test can prove the behavior is correct if I understood correctly:
```
  @Test
  void test(VertxTestContext testContext) {
    webClient.post("/quotes")
      .expect(ResponsePredicate.SC_BAD_REQUEST)
      .sendJsonObject(new JsonObject().put("author", "text"),
        testContext.succeeding(response -> {
          testContext.verify(() -> {
            assertEquals(400, response.statusCode(), response.bodyAsString());
            testContext.completeNow();
          });
        }));
  }
```

Reply:
You are right, I misread the change. Can you add this test to your PR?